lib/pavilion/schedulers/plugins/slurm.py

```diff
@@ -186,6 +188,80 @@ def slurm_states(state):
 
     return states
 
+def parse_node_list_element(node_spec: str):
+    """ Takes a node_spec, a string with a node name or a prefix and a
+    shorthand list of nodes in [] brackets.
+    Returns a fully resolved list of nodes.
+    """
+
+    # No shorthand list, node_spec is fully resolved.
+    bracket_check = ("[" in node_spec) + ("]" in node_spec)
+    if bracket_check == 0:
+        return [node_spec]
+    elif bracket_check == 1:
+        raise ValueError(
+            "Invalid Node List: '{}'. Unmatched range bracket []"
+            .format(node_spec)
+        )
+
+    # Get the node prefix by splitting on the starting square bracket.
+    # Strip the end bracket and split into a list of strings on comma.
+    # If the string has a dash it's a range, otherwise it's a single node.
+    # Remember the length of the first node number string in the list.
+    # Make the numbers into integers, create ranges and collect them as strings.
+    isplit = node_spec.split("[")
+    prefix = isplit[0]
+    nums = isplit[-1].strip("]").split(',')
+    nlims = nums[0].split('-')
+    lennum = len(nlims[0])
+    nodenums = []
+    char_error = 0
+    for num in nums:
+        num2 = num.replace('-','')
+        if not num[-1].isdigit() or num2.isalnum():
+            char_error = num
+            break
+        if '-' in num:
+            numsplit = num.split('-')
+            if not numsplit[0][-1].isdigit():
+                char_error = num
+                break
+            nlims = [int(n) for n in numsplit]
+            if nlims[1] > nlims[0]:
+                raise ValueError(
+                    "In node list '{}' part '{}', node range ends "
+                    "before it starts."
+                    .format(node_spec, num))
+
+            nrng  = range(nlims[0], nlims[1]+1)
+            nodenums.extend([str(na) for na in nrng])
+        else:
+            nodenums.append(num)
+
+    # If the loop was broken for trailing nonsense. raise value error.
+    if char_error:
+        raise ValueError(
+            "Node spec element {} constains invalid characters."
+            .format(char_error)
+        )
```

Comment:
There's only one place you need to check for this, and the exception should be thrown there.